woocommerce/woocommerce-gutenberg-products-block
# Add update_customer_from_request action to Checkout flow

Part of https://github.com/woocommerce/woocommerce-blocks/issues/5457
Related to https://github.com/Automattic/woocommerce-subscriptions-core/pull/184
Which is related to https://github.com/woocommerce/woocommerce-subscriptions/pull/4384


This PR adds the `woocommerce_store_api_checkout_update_customer_from_request` action to Checkout flow, the action is very similar to `woocommerce_store_api_cart_update_customer_from_request` and I initially wanted to reuse the name but they have different flows so it didnt't make sense yet.

This action is needed so WooCommerce Subscription can hook into it to update the parent subscription.

### Testing instructions
Testing steps are a bit convoluted because you need to jump between 3 repos.
- Clone this PR from WooCommerce Subscriptions https://github.com/woocommerce/woocommerce-subscriptions/pull/4384
- Run `npm install` `composer install` `npm run build`
- Clone this PR from WooCommerce Subscriptions core https://github.com/Automattic/woocommerce-subscriptions-core/pull/184
- Symlink the subs core repository to the vendor/woocommerce folder of WooCommerce subscriptions `ln -s ~/path-to-subscription-core-local-clone ~/path-to-subscription/vendor/woocommerce/subscriptions-core`
- Create a subscription product.
- Add item to your cart, checkout.
- Go to WooCommerce -> Subscriptions -> new created subscription.
- Mind the address there, on the right, Subscriptions actions -> create pending renewal order.
- On My Account -> Orders you will see the new order, click Pay.
- On Checkout page, change your name or address, mind the new address.
- Place the order.
- Visit the Subscription page again, the address should have updated.

### Changelog entry

> Introduce new `woocommerce_store_api_checkout_update_customer_from_request` action.
